src/jit/lower.cpp

```diff
@@ -1546,7 +1546,7 @@ void Lowering::LowerCall(GenTree* node)
             }
 #endif
 #if defined(_TARGET_AMD64_) && defined(FEATURE_READYTORUN_COMPILER)
-            if (call->gtEntryPoint.addr != nullptr)
+            if ((call->gtEntryPoint.addr != nullptr) && !call->IsFastTailCall())
```

Comment:
The right #ifdef here is

#if defined(FEATURE_FASTTAILCALL) && defined(FEATURE_READYTORUN_COMPILER)

It just so happens right now, Amd64 is the only target on which fast tail call feature is turned ON.